### PR TITLE
Ben/dev 464 - Free allotment for New Users

### PR DESCRIPTION
**Pull request by bLopata**
Adds header for unsubscribed users showing the current message count. Sets up a trial membership valid for 1 year when the user loads the page. Decrements using protected server/admin method hooking to prevent injection attacks.

image

Trial subscription metadata 👇 
https://app.screencast.com/rpJGmIWHW7Ori

### CODE REVIEW

**Review by VVoruganti — CHANGES_REQUESTED** [review]
You also need to change the api/ folder to check for valid subscription or a remaining trial.

Should be in the `/api/security` file

**Comment by bLopata** [comment]
> You also need to change the api/ folder to check for valid subscription or a remaining trial.
> 
> Should be in the `/api/security` file

Good call. Thinking something like this:
` sub = (
        supabase.table("subscriptions")
        .select("*")
        .eq("user_id", user.id)
        .or_(
            "status.eq.active,"  # Active subscriptions
            "and(status.eq.trialing,trial_end.gte.now())"  # Valid trial subscriptions
        )
        .execute()
    )`

unless you think we should have this call check the freeMessage allotment as well? Didn't know if that would be too tightly coupled.

File: www/utils/supabase/admin.ts [hunk 1]
```diff
@@ -21,6 +22,71 @@ const supabaseAdmin = createClient<Database>(
...
+    id: `free_trial_${userId}`,
+    user_id: userId,
+    status: 'trialing',
+    metadata: { freeMessages: FREE_MESSAGE_LIMIT },
+    price_id: null, // or your free tier price ID if you have one
+    quantity: 1,
+    cancel_at_period_end: false,
+    created: new Date().toISOString(),
+    current_period_start: new Date().toISOString(),
+    current_period_end: new Date(Date.now() + 365 * 24 * 60 * 60 * 1000).toISOString().replace('T', ' ').replace('Z', ''), // 1 year
```
Comment (VVoruganti) [inline www/utils/supabase/admin.ts:51]: Need to think through if it makes sense to send an end date for a year or just have no end date

File: www/utils/supabase/admin.ts [hunk 2]
```diff
@@ -21,6 +22,71 @@ const supabaseAdmin = createClient<Database>(
...
+  if (subError) throw new Error(`Subscription lookup failed: ${subError.message}`);
+
+  // If trial subscription exists, return it
+  if (existingSub) return existingSub;
+
+  // Create a new trial subscription
+  const subscriptionData: TablesInsert<'subscriptions'> = {
+    id: `free_trial_${userId}`,
+    user_id: userId,
+    status: 'trialing',
```
Comment (VVoruganti) [inline www/utils/supabase/admin.ts:44]: By manually inserting a subscription to the database we have a weird situation where there is no data in stripe linking the customer and Bloom user. Also our subscription check will get one subscription and if a user pays for a subscription, but also has a trialing subscription it'll lead to weird behavior.

File: www/utils/supabase/admin.ts [hunk 3]
```diff
@@ -21,6 +22,71 @@ const supabaseAdmin = createClient<Database>(
...
+
+  const currentCount = (subscription.metadata as { freeMessages: number })?.freeMessages ?? 0;
+  if (currentCount <= 0) return false;
+
+  const { error: updateError } = await supabaseAdmin
+    .from('subscriptions')
+    .update({
+      metadata: { freeMessages: currentCount - 1 }
+    })
+    .eq('id', subscription.id);
```
Comment (VVoruganti) [inline www/utils/supabase/admin.ts:83]: Can combine this into a single database operation example from claude:

```typescript
const { error: updateError } = await supabaseAdmin
  .from('subscriptions')
  .update({
    'metadata->freeMessages': 'metadata->freeMessages - 1'
  })
  .eq('id', subscription.id)
  .gt('metadata->freeMessages', 0); // Only update if freeMessages > 0

// Check if any rows were affected
if (updateError) return false;
```

**Review by VVoruganti — CHANGES_REQUESTED** [review]

**Pull request merged**